### PR TITLE
community: Add function call support in Tongyi chat model.

### PR DESCRIPTION
**Pull request by pcliupc**
- [ ] **PR message**: 
    - **Description:** This pr adds function calling support in Tongyi chat model.
    - **Issue:** None
    - **Dependencies:** None
    - **Twitter handle:** None

### CODE REVIEW

**Comment by pcliupc** [comment]
@baskaryan Can you please help review?

**Comment by pcliupc** [comment]
#20450

**Review by liugddx — APPROVED** [review]
LGTM

File: libs/community/langchain_community/chat_models/tongyi.py [hunk 1]
```diff
@@ -71,8 +71,14 @@ def convert_dict_to_message(
...
     elif role == "assistant":
+        if "tool_calls" in _dict:
+            additional_kwargs = {"tool_calls": _dict.get("tool_calls", [])}
+        else:
+            additional_kwargs = {}
         return (
-            AIMessageChunk(content=content) if is_chunk else AIMessage(content=content)
+            AIMessageChunk(content=content)
+            if is_chunk
+            else AIMessage(content=content, additional_kwargs={**additional_kwargs})
```
Comment (baskaryan) [inline libs/community/langchain_community/chat_models/tongyi.py:81]: could we update to also set the new AIMessage.tool_calls param? can see ChatOpenAI for an example

Comment (pcliupc) [reply libs/community/langchain_community/chat_models/tongyi.py]: @baskaryan fixed according to comments.

**Review by baskaryan — APPROVED** [review]

**Pull request merged**